packages/worker/src/workflow/threaded-vm.ts

```diff
@@ -74,21 +79,25 @@ export class WorkerThreadClient {
 
       completion.resolve(result.output);
     });
-    workerThread.on('exit', () => {
+    workerThread.on('error', (err) => {
+      logger.error(`Workflow Worker Thread failed: ${err}`, err);
```

Comment:
Can this fire multiple times, or it is just once? If the former you may want to lock the first exitError...

Reply:
Node.js' doc is not clear on guarantees here, but I don't think we could possibly receive multiple `error` events. And assuming we would, that would likely indicate that prior errors were not fatal, and consequently, we might really be interested only in the _latest_ error, not the first.